lib/template-compiler-plugin.d.ts

```diff
@@ -0,0 +1,23 @@
+import Filter = require('broccoli-persistent-filter');
```

Comment:
I do not want to ship ambient module declarations for modules that we actually provide (the ambient module declarations that we have at the moment are strictly for the fabricated `hbs` entry point). I'm happy to entertain a refactor to leverage TS itself as a separate PR but that is unrelated to the work being done in this PR. When we do that, we'll have to nail down the proper SemVer strategy.

Reply:
I removed this types file.